### DISCUSSION TITLE
Relationships keep disappearing when re-fetching database tables

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2373] [Ideas]
I there a way I can make the Relationships I have created from my SQL tables stay when I refresh the tables? when I add a table to the DB or when I add a column to a table. I have to "fetch tables from DB" and that wipes out all of my relationships and display columns. is there a way I can add new things to my DB and import the new data to BB without rebuilding the data structure each time? 

any help would be great thank you.

**Comment by mjashanks** [comment]
Hey Andrew - I can see why this is very annoying. I can also see why this might be complex to implement, as we would have to detect changes and decide what to do when foreign key columns disappear/get named.

I'm converting this to an idea and leaving it here so we can pick it up again when possible. There's no solution for it at the moment, unfortunately.

**Comment by shogunpurple** [comment]
I've created an issue for this here - the relationships should not be overridden so it's a bug: https://github.com/Budibase/budibase/issues/2377

**Comment by KernOil** [comment]
@mjashanks @shogunpurple  Thank you guys!
